### DISCUSSION TITLE
Component testing - run only a single test in the UI

### DISCUSSION THREAD

**Discussion by gsouf** [cypress-io/cypress#23669] [General]
Hi,

Recently I'm using cypress instead of storybook for developing components in isolation. That's a much better solution in my opinion. However there is something that slows me down during the development phase.. I have a spec file with multiple tests inside. I open the cypress UI to visualize changes I make on my component when I save the file. However everytime I save the file, every single tests will run but most of the times I want only one of the tests in my spec file to run as it's much faster that running everything everything. In other term I want to have instant feedback rather than looking at all my test running.

With an example. Given I have the following `Something.cy.js` file:

```
describe('Something', () => {

   it('does something', () => {  ....  });

   it('does something else', () => {  ....  });

});
```

In the **cypress UI** when I open the spec `Something` in **component testing** I would like to run **only** ` it('does something else', () => {  ....  });` and **not any** of the the other `it`.

### So my question is:

Is there a way I run only one test from a spec file in the UI? If not, is there any chance to see such a feature implemented?

**Comment by lmiller1990** [comment]
I agree, we should have this. It's a duplicate of: https://github.com/cypress-io/cypress/issues/23676

**Reply by gsouf** [reply]
Thanks for checking!
